lib/edit-site-page.php

```diff
@@ -146,49 +145,44 @@ function gutenberg_edit_site_init( $hook ) {
 		$settings['fontSizes'] = $font_sizes;
 	}
 
-	// Get all templates by triggering `./template-loader.php`'s logic.
-	$template_getters  = array(
-		'get_embed_template',
-		'get_404_template',
-		'get_search_template',
-		'get_home_template',
-		'get_privacy_policy_template',
-		'get_post_type_archive_template',
-		'get_taxonomy_template',
-		'get_attachment_template',
-		'get_single_template',
-		'get_page_template',
-		'get_singular_template',
-		'get_category_template',
-		'get_tag_template',
-		'get_author_template',
-		'get_date_template',
-		'get_archive_template',
+	$template_types = array(
+		'index',
+		'404',
+		'archive',
+		'author',
+		'category',
+		'tag',
+		'taxonomy',
+		'date',
+		// Skip 'embed' for now because it is not a regular template type.
+		'home',
+		'front-page',
+		'privacy-policy',
+		'page',
+		'search',
+		'single',
+		'singular',
+		'attachment',
 	);
+
 	$template_ids      = array();
 	$template_part_ids = array();
-	foreach ( $template_getters as $template_getter ) {
-		call_user_func( $template_getter ); // This sets $_wp_current_template_hierarchy.
-
-		$current_template_post = gutenberg_find_template_post( $_wp_current_template_hierarchy );
+	foreach ( $template_types as $template_type ) {
+		$template_hierarchy    = get_template_hierachy( $template_type );
+		$current_template_post = gutenberg_find_template_post( $template_hierarchy );
 		if ( isset( $current_template_post ) ) {
 			$template_ids[ $current_template_post->post_name ] = $current_template_post->ID;
 		}
 		if ( isset( $_wp_current_template_part_ids ) ) {
 			$template_part_ids = $template_part_ids + $_wp_current_template_part_ids;
 		}
 
-		$_wp_current_template_hierarchy = null;
-		$_wp_current_template_part_ids  = null;
-	}
-	get_front_page_template();
-	get_index_template();
-	$current_template_post                             = gutenberg_find_template_post( $_wp_current_template_hierarchy );
-	$template_ids[ $current_template_post->post_name ] = $current_template_post->ID;
-	if ( isset( $_wp_current_template_part_ids ) ) {
-		$template_part_ids = $template_part_ids + $_wp_current_template_part_ids;
+		$_wp_current_template_part_ids = null;
 	}
-	$settings['templateId']      = $current_template_post->ID;
+
+	$current_template_id = $template_ids['index'] ?? $template_ids['front-page'];
```

Comment:
I don't think we should load both the `index` template and the `front-page` template under any circumstances.

I mirrored Core's `template-loader` here:

https://github.com/WordPress/WordPress/blob/8ca76cabfc9caf75637f0e7f2689c40a0f184657/wp-includes/template-loader.php#L93-L95